### PR TITLE
Implement looser definition of equality

### LINKED ISSUE #80: Implement looser definition of equality for entity

### PR DESCRIPTION
**Pull request by hcwong**
## Description

This fixes #80, by implementing a non id based definition of equality

## Checks
- [x] No errors when running the tests
- [x] Build and checkstyle passes
- [] Run the app and ran 2 commands without failing
- [x] Written the baseline test cases for the PR

## Changelog
- [x] Changed all the update metods and the way they detect equality
- [x] Wrote tests

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -167,13 +167,13 @@ public void updateParticipant(Id id, Participant participant) throws AlfredExcep
             this.participantList.update(id, participant);
             return;
         }
+        this.participantList.update(id, participant);
+
         boolean isSuccessful = targetTeam.updateParticipant(participant);
         if (!isSuccessful) {
             logger.warning("The participant is not in the team provided");
```
Comment (john0227) [inline src/main/java/seedu/address/model/ModelManager.java]: If you get the team by participant id, why would the participant not be in that team? And so we're allowing the user to update the participant even if he/she's not in a team right?

File: src/main/java/seedu/address/model/entity/Participant.java [hunk 2]
```diff
@@ -124,8 +124,8 @@ public boolean isSameParticipant(Participant otherParticipant) {
             return true;
         }
 
-        return this.id.equals(otherParticipant.getId())
-                && (this.name.equals(otherParticipant.getName())
-                    || this.email.equals(otherParticipant.getEmail()));
+        return this.name.equals(otherParticipant.getName())
+                || (this.phone.equals(otherParticipant.getPhone())
```
Comment (john0227) [inline src/main/java/seedu/address/model/entity/Participant.java]: I think `&&` and `||` are switched.

File: src/main/java/seedu/address/model/entity/Team.java [hunk 3]
```diff
@@ -325,6 +325,6 @@ public boolean isSameTeam(Team otherTeam) {
         }
 
         return this.name.equals(otherTeam.getName())
-                || this.id.equals(otherTeam.getId());
+                || this.projectName.equals(otherTeam.getProjectName());
```
Comment (john0227) [inline src/main/java/seedu/address/model/entity/Team.java]: I think this should be `&&`

File: src/test/java/seedu/address/model/entitylist/MentorListTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import org.junit.jupiter.api.TestInstance;
+
+import seedu.address.commons.exceptions.AlfredModelException;
+import seedu.address.model.entity.Id;
+import seedu.address.model.entity.Mentor;
+import seedu.address.model.entity.Name;
+import seedu.address.model.entity.PrefixType;
+import seedu.address.testutil.TypicalMentors;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (john0227) [inline src/test/java/seedu/address/model/entitylist/MentorListTest.java]: This one I'm just curious as to what it does lol

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (hcwong) [reply src/main/java/seedu/address/model/ModelManager.java]: For the first one, maybe the participant does not belong to a team or has not been assigned. And yes to the second, we always allow for the participant in the participant list to be updated

File: src/main/java/seedu/address/model/entity/Team.java (hunk 3 above)
Comment (hcwong) [reply src/main/java/seedu/address/model/entity/Team.java]: I was thinking that both should also be unique, else it is possible for a team to have two different projects or for two different teams to have the same project name

File: src/test/java/seedu/address/model/entitylist/MentorListTest.java (hunk 4 above)
Comment (hcwong) [reply src/test/java/seedu/address/model/entitylist/MentorListTest.java]: This is needed so I can run the @BeforeAll signature to make sure that the list is set to 0 before the start of each test if not my assert statements might gg 💣

File: src/main/java/seedu/address/model/entity/Team.java (hunk 3 above)
Comment (hcwong) [reply src/main/java/seedu/address/model/entity/Team.java]: what do you think?

File: src/main/java/seedu/address/model/entity/Participant.java (hunk 2 above)
Comment (hcwong) [reply src/main/java/seedu/address/model/entity/Participant.java]: yep thanks for the catch

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (john0227) [reply src/main/java/seedu/address/model/ModelManager.java]: Cool I was just wondering about `if (!isSuccessful)` part

File: src/main/java/seedu/address/model/entity/Team.java (hunk 3 above)
Comment (john0227) [reply src/main/java/seedu/address/model/entity/Team.java]: So it should be && right?

File: src/test/java/seedu/address/model/entitylist/MentorListTest.java (hunk 4 above)
Comment (john0227) [reply src/test/java/seedu/address/model/entitylist/MentorListTest.java]: oh..not sure if i understand but cool lol

File: src/main/java/seedu/address/model/entity/Team.java (hunk 3 above)
Comment (hcwong) [reply src/main/java/seedu/address/model/entity/Team.java]: Oh but if is && then both must be the same to be equal, but either or should be unique?

File: src/test/java/seedu/address/model/entitylist/MentorListTest.java (hunk 4 above)
Comment (hcwong) [reply src/test/java/seedu/address/model/entitylist/MentorListTest.java]: BeforeAll is to signal that the function is a setup method that should be run before all tests. Quite useful. But javafx was complaining that I had to put this `@TestInstance` thing to use BeforeAll so i added it there

**Pull request merged**